### DISCUSSION TITLE
Add ability to assign a Prefix to a L2VPN

### DISCUSSION THREAD

**Discussion by v0tti** [netbox-community/netbox#12827] [Ideas]
Currently it is only possible to assign a prefix to a VLAN. The ability to assign them to L2VPNs in the same manner should be added. 

We are not assigning prefixes to VLANs directly in our network. VLANs are only terminations of L2VPNs (in our case mostly VPLS). In the case where we only terminate a L2VPN to one VLAN this works fine (we just assign the prefixes to that VLAN), but it gets complicated if we have multiple terminations to different VLANs.
We could use custom fields to add prefixes to L2VPNs, but that would not allow us to see the L2VPN membership of a prefix.

Does this make sense and is there any chance of this being implemented into NetBox?

**Comment by mtinberg** [comment]
As a general principal, if this doesn't get immediately picked up and added to the built-in data models, I highly encourage you to invest in using custom object fields and custom urls (to template a search for the reverse relationship) to implement the relationship you want, so you can start getting value from the data right away.  The bar is very low for adding custom fields and you get like 90% of the benefit with much less risk/cost than making code changes.

A major benefit of building data models now using custom fields/urls is getting the operational/user experience right away which can be useful when designing built-in support or a plugin later.  If the shape of some data or UI isn't quite right, fixing it in your netbox is easier than needing to fix it in everyone's netbox later.


—

Mark Tinberg ***@***.***>

Division of Information Technology-Network Services

University of Wisconsin-Madison


________________________________
From: Alexander Votteler ***@***.***>
Sent: Wednesday, June 7, 2023 4:58 AM
To: netbox-community/netbox ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [netbox-community/netbox] Add ability to assign a Prefix to a L2VPN (Discussion #12827)


Currently it is only possible to assign a prefix to a VLAN. The ability to assign them to L2VPNs in the same manner should be added.

We are not assigning prefixes to VLANs directly in our network. VLANs are only terminations of L2VPNs (in our case mostly VPLS). In the case where we only terminate a L2VPN to one VLAN this works fine (we just assign the prefixes to that VLAN), but it gets complicated if we have multiple terminations to different VLANs.
We could use custom fields to add prefixes to L2VPNs, but that would not allow us to see the L2VPN membership of a prefix.

Does this make sense and is there any chance of this being implemented into NetBox?

**Reply by v0tti** [reply]
Thanks. Sure, I will be implementing this using custom fields in my NetBox soon if this does not get support by the maintainers. I just thought it is worth starting a discussion on it since it seems strange that this option is missing.
